src/bot.ts

```diff
@@ -520,10 +524,12 @@ export class DiscordBot {
             });
           });
         });
-      });
-      if (msg.content !== null && msg.content !== "") {
-        this.msgProcessor.FormatDiscordMessage(msg).then((result) => {
-            rooms.forEach((room) => {
+      }).then(() => {
+        if (msg.content === null || msg.content === "") {
+          return null;
+        }
+        return this.msgProcessor.FormatDiscordMessage(msg).then((result) => {
+            return Bluebird.map(rooms, (room: string) => {
```

Comment:
What's wrong with `this.SendMatrixMessage`?

Reply:
I don't know. I didn't write that piece of code. I just changed its surroundings.
Just looking at it it seems like SendMatrixMessage assumes the user to have already joined the room while this code doesn't.